init.d/net.lo.in

```diff
@@ -118,6 +118,31 @@ _flatten_array()
 	_array_helper $1
 }
 
+_wait_for_presence()
+{
+	local timeout=
+
+	_exists && return 0
+
+	eval timeout=\$presence_timeout_${IFVAR}
+	timeout=${timeout:-${presence_timeout:-0}}
+
+	[ ${timeout} -le 0 ] && return 1
+
+	einfon "Waiting for ${IFACE} to show up (${timeout} seconds)"
+	while [ ${timeout} -gt 0 ]; do
+		_exists && break
```

Comment:
if you capture the return code here, then the final _exists call is not needed (can also change this to be a do/while loop for a similar effect)

Reply:
If we want to change this "break" to "return" then we'll also need to add "echo" and "eend 0"
before such "return".
No problem doing that, but I think a common "echo" and "eend" at the function end just looks
nicer due to less code duplication.